src/util/worker/mod.rs

```diff
@@ -88,6 +91,10 @@ pub trait BatchRunnable<T: Display> {
     fn shutdown(&mut self) {}
 }
 
+pub trait BatchRunnableWithTimer<T: Display, U>: BatchRunnable<T> {
+    fn on_timeout(&mut self, TimeoutTask<U>, &mut Timer<U>);
```

Comment:
I think only `U` and `&mut Timer<U>` are necessary.

Reply:
If you want to add the task back into the `timer`, we can get the `timeout` from `TimeoutTask`, or we need to remember the `timeout` in some other places.

Reply:
I think it is fine here, we may need to add other timers in the callback like mio does.

But, I prefer to exchange the place of TimeoutTask and Timer.

Reply:
This is timeout style rather then interval, so task is fired then forget. If caller wants to fire it again, then it's caller's responsibility to store the timeout of task.

Reply:
yes, I am wrong. I thought TimeoutTask is task, but it’s not. Btw, I still think Timer should be in front of U

Reply:
yes, I am wrong. I thought TimeoutTask is task, but it’s not. Btw, I still think Timer should be in front of U

Reply:
yes, I am wrong. I thought TimeoutTask is task, but it’s not. Btw, I still think Timer should be in front of U

Reply:
got it.